src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -891,6 +992,27 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_in(input_ids)
 
+        # Attention mask.
+        if attention_mask is not None:
+            assert batch_size > 0, "batch_size has to be defined and > 0"
+            attention_mask = attention_mask.view(batch_size, -1)
+            if self._use_flash_attention_2:
+                attention_mask = attention_mask if 0 in attention_mask else None
+            elif self._use_sdpa and not output_attentions:
+                attention_mask = _prepare_4d_causal_attention_mask_for_sdpa(
+                    attention_mask=attention_mask,
+                    input_shape=(batch_size, seq_length),
+                    inputs_embeds=inputs_embeds,
+                    past_key_values_length=past_length,
+                )
+            else:
+                attention_mask = _prepare_4d_causal_attention_mask(
```

Comment:
Exchanged the old attention mask implementation here. Can revert but thought it kept things cleaner.